### PR TITLE
Add a nonuniform attribute silencing warning on coercions

### PR DESCRIPTION
**Pull request by proux01**
This is a follow up of https://github.com/coq/coq/pull/15789 implementing a comment of @CohenCyril on Zulip: https://coq.zulipchat.com/#narrow/stream/237977-Coq-users/topic/Identity.20Coercions/near/276371166

The #[nonuniform] boolean attribute helps silencing the non-uniform-inheritance warning on coercions when the user needs to declare such a coercion on purpose.

- [x] Added / updated **test-suite**.
- [x] Added **changelog**.
- [x] Added / updated **documentation**.

#### Overlays (to be merged synchronously with the PR)
* https://github.com/LPCIC/coq-elpi/pull/352
* https://github.com/Mtac2/Mtac2/pull/355

### CODE REVIEW

File: vernac/vernacentries.ml
Comment (SkySkimmer) [inline vernac/vernacentries.ml:68]:
```suggestion
    let nonuniform = if coercion then ComCoercion.nonuniform else Notations.return None in
    let ((((((locality, deprecated), polymorphic), program), canonical_instance), typing_flags), using), nonuniform =
      parse Notations.(locality ++ deprecation ++ polymorphic ++ program ++ canonical_instance ++ typing_flags ++ using ++ nonuniform) f
```
and remove the `if not coercion` check below

File: vernac/vernacentries.ml
Comment (SkySkimmer) [inline vernac/vernacentries.ml:95]:
```suggestion
let with_def_attributes ?coercion ~atts f =
  let atts = DefAttributes.parse ?coercion atts in
```

File: vernac/vernacentries.ml
Comment (proux01) [reply vernac/vernacentries.ml]: Nice, thanks!

File: vernac/vernacentries.ml
Comment (proux01) [reply vernac/vernacentries.ml:68]: Ok, this file has many very long lines anyway

File: doc/sphinx/addendum/implicit-coercions.rst
Comment (jfehrle) [inline doc/sphinx/addendum/implicit-coercions.rst:126]:
```suggestion
  This command supports the :attr:`local` and :attr:`global` attributes
  (described :ref:`here <set_unset_scope_qualifiers>`) as well as the
  :attr:`nonuniform` attribute.
```

File: doc/sphinx/addendum/implicit-coercions.rst
Comment (jfehrle) [inline doc/sphinx/addendum/implicit-coercions.rst:130]:
```suggestion
     Silence the non uniform inheritance warning.
```

I think "see below" is not helpful here because the warning message just repeats what's said here.
Furthermore the target is half a page below and a bit buried.

**Review by jfehrle — COMMENTED** [review]
A couple suggestions but looks good.

**Comment by proux01** [comment]
This is looking for an assignee, maybe @ana-borges since you somewhat requested the feature? (I don't think you have push rights, but maybe we can ask)

**Comment by ana-borges** [comment]
Sure! Indeed I don't think I have the rights to merge, unless that comes with being a member of the Coq organization, which I already am.

**Comment by proux01** [comment]
Let's ask @Zimmi48 , I think you should have a look to the maintainer teams: https://github.com/orgs/coq/teams/contributors/teams
and tell him to which one you'd like to be added (for instance number-maintainers contains Sint63).

**Comment by ana-borges** [comment]
I guess the bedrock2 failure is unrelated. @SkySkimmer are you done reviewing?

**Comment by SkySkimmer** [comment]
yes

**Comment by ana-borges** [comment]
@coqbot: merge now

**Comment by ana-borges** [comment]
Right, so @SkySkimmer can you please approve?

**Comment by SkySkimmer** [comment]
Why not do it yourself?

**Comment by ana-borges** [comment]
Seemed to make more sense for the person who actually reviewed to approve and I didn't want to overstep. Consider my notion of the reviewing culture duly updated.

**Review by ana-borges — APPROVED** [review]

**Comment by ana-borges** [comment]
@coqbot: merge now

**Comment by SkySkimmer** [comment]
I didn't review in depth, I just had a look.

**Comment by proux01** [comment]
Thanks everybody!

**Pull request merged**
